LayoutTests/media/media-source/media-source-video-playback-quality.html

```diff
@@ -60,10 +62,18 @@
         run('sourceBuffer.appendBuffer(samples)');
     }
 
-    function samplesAdded()
+    async function samplesAdded()
+    {
+        run('source.endOfStream()');
+        run('video.play()');
+        await waitFor(video, 'ended');
+        setTimeout(videoEnded, 50); // VideoPlaybackQuality is refreshed every 0.25s, wait a bit.
```

Comment:
Isn't there a risk for flakiness?
Should we do the try test and if failing retry until 100 times approach?

Reply:
I'm not familiar with this. What if 100 loops is done under 0.25s?

Reply:
Actually, with our current implementation there's no risk of flakiness:
- With WK1 or WK2 with MockMSE in the CP, the object is updated synchronously when you call `getVideoPlaybackQuality()`
- With the MockMSE in the GPU Process, the VPQ is updated in the next event loop
```
void RemoteMediaPlayerProxy::updateCachedState(bool forceCurrentTimeUpdate)
{
    if (!m_observingTimeChanges || forceCurrentTimeUpdate)
        currentTimeChanged(m_player->currentTime());

    m_cachedState.paused = m_player->paused();
    maybeUpdateCachedVideoMetrics();
    if (m_bufferedChanged) {
        m_bufferedChanged = false;
        if ((m_engineIdentifier != MediaPlayerEnums::MediaEngineIdentifier::AVFoundationMSE)
            && (m_engineIdentifier != MediaPlayerEnums::MediaEngineIdentifier::MockMSE)) {
            m_cachedState.bufferedRanges = *m_player->buffered();
        }
    }
}
```
it sends the time , and immediately after the player state, which contains the VPQ